examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp

```diff
@@ -89,6 +93,182 @@ CHIP_ERROR EVSEManufacturer::Shutdown()
     return CHIP_NO_ERROR;
 }
 
+/**
+ * @brief   Allows a client application to send in power readings into the system
+ *
+ * @param[in]  aEndpointId     - Endpoint to send to EPM Cluster
+ * @param[in]  aActivePower_mW - Power measured in milli-watts
+ * @param[in]  aVoltage_mV     - Voltage measured in milli-volts
+ * @param[in]  aCurrent_mA     - Current measured in milli-amps
+ */
+CHIP_ERROR EVSEManufacturer::SendPowerReading(EndpointId aEndpointId, int64_t aActivePower_mW, int64_t aVoltage_mV,
+                                              int64_t aCurrent_mA)
+{
+    // TODO add Power Readings when EPM cluster is merged
+
+    return CHIP_NO_ERROR;
+}
+
+/**
+ * @brief   Allows a client application to send in energy readings into the system
+ *
+ *          This is a helper function to add timestamps to the readings
+ *
+ * @param[in]  aCumulativeEnergyImported -total energy imported in milli-watthours
+ * @param[in]  aCumulativeEnergyExported -total energy exported in milli-watthours
+ */
+CHIP_ERROR EVSEManufacturer::SendEnergyReading(EndpointId aEndpointId, int64_t aCumulativeEnergyImported,
+                                               int64_t aCumulativeEnergyExported)
+{
+    MeasurementData * data = MeasurementDataForEndpoint(aEndpointId);
+
+    EnergyMeasurementStruct::Type energyImported;
+    EnergyMeasurementStruct::Type energyExported;
+
+    uint32_t oldTimestamp;
+
+    // Get current timestamp
+    uint32_t currentTimestamp;
+    CHIP_ERROR err = GetEpochTS(currentTimestamp);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "GetEpochTS returned error getting timestamp");
+        return err;
+    }
+
+    /** IMPORT */
+    // Copy last endTimestamp into new startTimestamp if it exists
+    energyImported.startTimestamp.ClearValue();
+    if (data->cumulativeImported.HasValue())
+    {
+        if (data->cumulativeImported.Value().endTimestamp.HasValue())
+        {
+            oldTimestamp = data->cumulativeImported.Value().endTimestamp.Value();
+            energyImported.startTimestamp.SetValue(oldTimestamp);
+        }
+    }
+
+    energyImported.endTimestamp.SetValue(currentTimestamp);
+    energyImported.energy = aCumulativeEnergyImported;
+
+    /** EXPORT */
+    // Copy last endTimestamp into new startTimestamp if it exists
+    energyExported.startTimestamp.ClearValue();
+    if (data->cumulativeExported.HasValue())
+    {
+        if (data->cumulativeExported.Value().endTimestamp.HasValue())
+        {
+            oldTimestamp = data->cumulativeExported.Value().endTimestamp.Value();
+            energyExported.startTimestamp.SetValue(oldTimestamp);
+        }
+    }
+    energyExported.endTimestamp.SetValue(currentTimestamp);
+    energyExported.energy = aCumulativeEnergyExported;
+
+    // call the SDK to update attributes and generate an event
+    NotifyCumulativeEnergyMeasured(aEndpointId, MakeOptional(energyImported), MakeOptional(energyExported));
```

Comment:
this seems to return a `bool`. If it returns a value, should it be handled?

Reply:
Looking at the impl, I wonder why it returns a value ... can the caller do anything about it (or is it expected to, like not resetting start/end times )? otherwise we may as well return void and just log failures.

Reply:
I added a return handler with a log output.